src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -1560,13 +1562,16 @@ private void addPBXTargetDependency(PBXNativeTarget pbxTarget, BuildTarget depen
       // we have a plain apple_binary that has Swift deps. So we're manually doing exactly what
       // Xcode does to make sure binaries link successfully if they use Swift directly or
       // transitively.
-      ImmutableSet<PBXFileReference> swiftDeps =
-          collectRecursiveLibraryDependenciesWithSwiftSources(node);
       if (swiftDeps.size() > 0) {
         librarySearchPaths.add("$DT_TOOLCHAIN_DIR/usr/lib/swift/$PLATFORM_NAME");
       }
     }
 
+    if (swiftDeps.size() > 0) {
+      ldRunpathSearchPaths.add("@executable_path/Frameworks");
```

Comment:
@robbertvanginkel This would not work for macOS, we will need to use `@executable_path/../Frameworks` (same for the line below).

Reply:
Pushed a fix for this.